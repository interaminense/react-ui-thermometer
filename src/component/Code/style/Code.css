.code {
  background-color: #333;
  padding: 5px 10px;
  border-radius: 5px;
  color: #fff;
  display: inline-block; }
  .code .code__tag-name {
    color: #4ec9b0; }
  .code .code__props {
    color: #9cdcfe; }
  .code .code__value-string {
    color: #ce9178; }
  .code .code__value-int {
    color: #b0cea8; }
