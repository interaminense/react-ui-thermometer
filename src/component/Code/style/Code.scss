$color-white: #fff;
$color-gray: #333;
$color-a: #4ec9b0;
$color-b: #9cdcfe;
$color-c: #ce9178;
$color-d: #b0cea8;

.code {
  background-color: $color-gray;
  padding: 5px 10px;
  border-radius: 5px;
  color: $color-white;
  display: inline-block;

  .code__tag-name {
    color: $color-a;
  }
  .code__props {
    color: $color-b;
  }
  .code__value-string {
    color: $color-c;
  }
  .code__value-int {
    color: $color-d;
  }
}