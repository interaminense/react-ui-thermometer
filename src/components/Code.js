import React, { Component } from 'react';
import '../styles/Code.scss';

class Code extends Component {
	constructor(props) {
		super(props);
		
		this.state = {
			arrowLeft: '<',
			arrowRight: '>',
			keyLeft: '{',
			keyRight: '} ',
			tagName: this.props.tagName || '',
			values: this.props.prop || []
		};
	}
	
	render() {
		
		const props = this.state.values.map((prop, i) => {
			return (
				<span key={i}>
				<span className="code__props">{prop.name}</span>={this.state.keyLeft}
				{typeof prop.val === 'string' ? (
					<span className={'code__value-string'}>'{prop.val}'</span>
					) : (
						<span className={'code__value-int'}>{prop.val}</span>
						)}
						{this.state.keyRight}
						</span>
						);
					});
					
					return (
						<pre className="code">
						{this.state.arrowLeft}
						<span className="code__tag-name">{this.state.tagName} </span>
						{props}
						/{this.state.arrowRight}
						</pre>
						);
					}
				}
				
				export default Code;