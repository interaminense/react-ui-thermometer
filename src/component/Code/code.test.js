import React from 'react';
import Code from './Code';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import { shallow } from 'enzyme';

configure({ adapter: new Adapter() });
describe('Code Component tests', () => {

    it('should render the component', () => {
        const wrapper = shallow(
            <Code
                tagName="name"
                prop={[
                    { name: 'theme', val: 'light' },
                    { name: 'value', val: 80 }]}
            />);
        expect(wrapper).toMatchSnapshot();
    });
});