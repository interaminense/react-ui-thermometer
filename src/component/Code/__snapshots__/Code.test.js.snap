// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code Component tests should render the component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Code
    prop={
      Array [
        Object {
          "name": "theme",
          "val": "light",
        },
        Object {
          "name": "value",
          "val": 80,
        },
      ]
    }
    tagName="name"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "<",
        <span
          className="code__tag-name"
        >
          name
           
        </span>,
        Array [
          <span>
            <span
              className="code__props"
            >
              theme
            </span>
            =
            {
            <span
              className="code__value-string"
            >
              '
              light
              '
            </span>
            } 
          </span>,
          <span>
            <span
              className="code__props"
            >
              value
            </span>
            =
            {
            <span
              className="code__value-int"
            >
              80
            </span>
            } 
          </span>,
        ],
        "/",
        ">",
      ],
      "className": "code",
    },
    "ref": null,
    "rendered": Array [
      "<",
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "name",
            " ",
          ],
          "className": "code__tag-name",
        },
        "ref": null,
        "rendered": Array [
          "name",
          " ",
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="code__props"
            >
              theme
            </span>,
            "=",
            "{",
            <span
              className="code__value-string"
            >
              '
              light
              '
            </span>,
            "} ",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "theme",
              "className": "code__props",
            },
            "ref": null,
            "rendered": "theme",
            "type": "span",
          },
          "=",
          "{",
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "'",
                "light",
                "'",
              ],
              "className": "code__value-string",
            },
            "ref": null,
            "rendered": Array [
              "'",
              "light",
              "'",
            ],
            "type": "span",
          },
          "} ",
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="code__props"
            >
              value
            </span>,
            "=",
            "{",
            <span
              className="code__value-int"
            >
              80
            </span>,
            "} ",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "value",
              "className": "code__props",
            },
            "ref": null,
            "rendered": "value",
            "type": "span",
          },
          "=",
          "{",
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": 80,
              "className": "code__value-int",
            },
            "ref": null,
            "rendered": 80,
            "type": "span",
          },
          "} ",
        ],
        "type": "span",
      },
      "/",
      ">",
    ],
    "type": "pre",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "<",
          <span
            className="code__tag-name"
          >
            name
             
          </span>,
          Array [
            <span>
              <span
                className="code__props"
              >
                theme
              </span>
              =
              {
              <span
                className="code__value-string"
              >
                '
                light
                '
              </span>
              } 
            </span>,
            <span>
              <span
                className="code__props"
              >
                value
              </span>
              =
              {
              <span
                className="code__value-int"
              >
                80
              </span>
              } 
            </span>,
          ],
          "/",
          ">",
        ],
        "className": "code",
      },
      "ref": null,
      "rendered": Array [
        "<",
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "name",
              " ",
            ],
            "className": "code__tag-name",
          },
          "ref": null,
          "rendered": Array [
            "name",
            " ",
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="code__props"
              >
                theme
              </span>,
              "=",
              "{",
              <span
                className="code__value-string"
              >
                '
                light
                '
              </span>,
              "} ",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "theme",
                "className": "code__props",
              },
              "ref": null,
              "rendered": "theme",
              "type": "span",
            },
            "=",
            "{",
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "'",
                  "light",
                  "'",
                ],
                "className": "code__value-string",
              },
              "ref": null,
              "rendered": Array [
                "'",
                "light",
                "'",
              ],
              "type": "span",
            },
            "} ",
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="code__props"
              >
                value
              </span>,
              "=",
              "{",
              <span
                className="code__value-int"
              >
                80
              </span>,
              "} ",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "value",
                "className": "code__props",
              },
              "ref": null,
              "rendered": "value",
              "type": "span",
            },
            "=",
            "{",
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": 80,
                "className": "code__value-int",
              },
              "ref": null,
              "rendered": 80,
              "type": "span",
            },
            "} ",
          ],
          "type": "span",
        },
        "/",
        ">",
      ],
      "type": "pre",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
